sdk/core/azure-core/src/http/request.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <string>
+
+#include <http/http.hpp>
+
+using namespace azure::core::http;
+using namespace std;
+
+std::string Header::getName() { return this->name; }
+std::string Header::getValue() { return this->value; }
+
+http_method::HttpMethod Request::getMethod() { return this->method; }
+string Request::getUrl() { return this->url; }
+string Request::getBody() { return this->body; }
+vector<Header> Request::getHeaders() { return this->headers; }
+
+void Request::addHeader(string name, string value)
+{
+  try
+  {
+    this->headers.push_back(Header(name, value));
+  }
+  catch (exception e)
+  {
+    throw e;
+  }
+}
+
+void Request::addQueryParameter(string name, string value)
+{
+  // Add question mark if there are not query parameters
+  if (this->query_start == 0)
+  {
+    this->url = this->url + "?";
+    this->query_start = this->url.length();
+  }
+  else
+  {
+    this->url = this->url + "&";
+  }
+
+  // adding name
+  this->url = this->url + name;
+  // Add symbol
+  this->url = this->url + "=";
+  // value
+  this->url = this->url + value;
```

Comment:
Values should be encoded before being appended to URL.

And this encoding rule is slightly different from regular encoding. I remember continuation token can appear as query parameter value in some API. The continuation token is base64-encoded string and may contain `=`. Although according to HTTP specification, `=` doesn't need to be encoded in this case, but storage server side is so strict that it just rejects such requests.

Reply:
yeah, url-encoding is still pending and will be added soon.
https://github.com/Azure/azure-sdk-for-cpp/issues/28

This is just the starting point and we want something with minimum functionality right :)